Mathlib/MeasureTheory/Integral/CircleAverage.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.MeasureTheory.Integral.CircleIntegral
+import Mathlib.MeasureTheory.Integral.IntervalAverage
+import Mathlib.MeasureTheory.Integral.IntervalIntegral.Periodic
+
+/-!
+# Circle Averages
+
+For a function `f` on the complex plane, this file introduces the notation `circleAverage f c R` as
+a shorthand for the average of `f` on the circle with center `c` and radius `R`. Averages of this
+form are typically used in analysis of one complex variable. Like `IntervalAverage`, this notion
+exists as a convenience. It avoids the hassle to manually elemininate `2 * π` every time an average
+is computed.
+
+Note: The relevant integrability property is `CircleIntegrable`, as defined in
+`Mathlib.MeasureTheory.Integral.CircleIntegral`.
+
+Implementation Note: Like `circleMap`, `circleAverage`s are defined for negative radii. The theorem
+`circleAverage_congr_negRadius` shows that the average is independent of the radius' sign.
+-/
+
+open Filter Metric Real
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜] [NormedSpace 𝕜 E] [SMulCommClass ℝ 𝕜 E]
+  {f f₁ f₂ : ℂ → E} {c : ℂ} {R : ℝ} {a : 𝕜}
+
+/-!
+# Definition
+-/
+
+variable (f c R) in
+/--
+Define `circleAverage f c R` as the average value of `f` on the circle with center `c` and radius
+`R`.
+-/
+noncomputable def circleAverage : E :=
```

Comment:
Are you sure that you want this formula? It's the integral with respect to `|dz|`, not with respect to `dz`. You can compare with the definition of `circleIntegral`, which involves an additional factor `deriv (circleMap c R) θ` . If this is really what you want, there should be big warnings everywhere (in the file-level docstring and in the docstring for the definition) saying that `circleIntegral` and `circleAverage` have a really different nature.

Reply:
My guess is that it's really what you want (add that you will apply this to harmonic functions, not holomorphic functions). Then I'd say this should be in the `Real` namespace (to differentiate this from a potential complex notion, which would involve `dz` instead of `|dz|`, and divide by `2i\pi` instead of `2\pi`). And big warnings in the docstrings!

Reply:
I edited the file-level docstring to clarify the role of `circleAverage`, and explain that this is a measure-theoretic average that should not be seen as a path integral.

@sgouezel I hope that this is what you had in mind?